src/plugins/charts/bar/BarGraphView.vue

```diff
@@ -297,25 +290,30 @@ export default {
 
             let xValues = [];
             let yValues = [];
-            const xAxisMetadata = axisMetadata.xAxisMetadata.find(metadata => metadata.source === this.domainObject.configuration.axes.xKey);
+            let xAxisMetadata = axisMetadata.xAxisMetadata.find(metadata => metadata.source === this.domainObject.configuration.axes.xKey);
             if (xAxisMetadata && xAxisMetadata.isArrayValue) {
                 //populate x and y values
                 let metadataKey = this.domainObject.configuration.axes.xKey;
                 if (data[metadataKey] !== undefined) {
-                    xValues = this.format(key, metadataKey, data);
+                    xValues = this.parse(key, metadataKey, data);
                 }
 
                 metadataKey = this.domainObject.configuration.axes.yKey;
                 if (data[metadataKey] !== undefined) {
-                    yValues = this.format(key, metadataKey, data);
+                    yValues = this.parse(key, metadataKey, data);
                 }
             } else {
                 //populate X and Y values for plotly
                 axisMetadata.xAxisMetadata.filter(metadataObj => !metadataObj.isArrayValue).forEach((metadata) => {
+                    if (!xAxisMetadata) {
+                        //Assign the first metadata to use for any formatting
+                        xAxisMetadata = metadata;
+                    }
+
                     xValues.push(metadata.name);
-                    if (data[metadata.key]) {
-                        const formattedValue = this.format(key, metadata.key, data);
-                        yValues.push(formattedValue);
+                    if (data[metadata.source]) {
```

Comment:
I don't _think_ we actually want to use the source. The `key` is what unique identifies a telemetry metadatum. The `source` defines where the data ultimately comes from, but when dealing with metadata and formatters we always want to use the key, not the source. 

Sorry, I didn't realize the context when you were asking about this earlier.

Reply:
Done